### DISCUSSION TITLE
Fetch total guild's before all shards start?

### DISCUSSION THREAD

**Discussion by callumok2004** [discordjs/discord.js#6893] [Q&A] [answered]
Is it possible to fetch the total guilds my bot is in before all shards have started?
And with cross-hosting, I want the total number of guilds for all shards before each shard has started.

**Comment by Error44s** [comment]
As far as I know it should work like this

```const Discord = require("discord.js");
const client = new Discord.Client({
    intents: ["GUILDS"],
});

client.on("ready", () => {
    const Guilds = client.guilds.cache.map(guild => guild.id);
    console.log(Guilds);
});

client.login(process.env.DISCORD_ACCESS_TOKEN);```

**Reply by callumok2004** [reply]
That will fetch the guilds for that shard, and that shard only, I want the total amount of guilds from all shards combined.

**Reply by Error44s** [reply]
```
shard.broadcastEval("this.guilds.map(u => u.id)").then(result => {
        for (var i = 0; i<result.length; i++){
            console.log(result[i]);
        }
    });
```

**Reply by callumok2004** [reply]
You cannot broadcast eval until all shards are ready.
" I want the total number of guilds for all shards **before each shard has started.**"

**Comment by KinectTheUnknown** [comment, marked as answer]
No, it’s not possible to obtain the number of guilds the bot is in before logging in. At most, you can make use of the [GET /gateway/bot](https://discord.com/developers/docs/topics/gateway#get-gateway-bot) endpoint, which will give you the recommended number of shards to spawn.
